### PR TITLE
Generic functions to print tables and lists

### PR DESCRIPTION
**Pull request by janelletavares**
## Description of change

Fixes https://github.com/meroxa/mdpx/issues/575

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

| before | after |
|--------|-------|
|||

### CODE REVIEW

File: utils/display/display.go [hunk 1]
```diff
@@ -9,3 +18,81 @@ func truncateString(oldString string, l int) string {
...
+	table := simpletable.New()
+	headers := []*simpletable.Cell{}
+	for column := range details {
+		headers = append(headers, &simpletable.Cell{Align: simpletable.AlignCenter, Text: strings.ToUpper(column)})
+	}
+	objs := interfaceSlice(input)
+	for _, o := range objs {
+		bytes, err := json.Marshal(o)
+		if err != nil {
+			fmt.Printf("err1: %v\n", err)
```
Comment (janelletavares) [inline utils/display/display.go:77]:
```suggestion
```

File: utils/display/display.go [hunk 2]
```diff
@@ -9,3 +18,80 @@ func truncateString(oldString string, l int) string {
...
+	objs := interfaceSlice(input)
+	for _, o := range objs {
+		bytes, err := json.Marshal(o)
+		if err != nil {
+			return ""
+		}
+		amorphous := map[string]interface{}{}
+		err = json.Unmarshal(bytes, &amorphous)
+		if err != nil {
+			fmt.Printf("err2: %v\n", err)
```
Comment (janelletavares) [inline utils/display/display.go:82]:
```suggestion
```

File: utils/display/environments_test.go
Comment (anna-cross) [inline utils/display/environments_test.go:1]: Do we need environments files? We won't have actual env logic in mdpx right?

Comment (janelletavares) [reply utils/display/environments_test.go:1]: i was under the impression that mdpx made environments easier and that we will continue to give customers to option to launch our platform in private environments. let's confirm.

File: utils/display/display.go [hunk 3]
```diff
@@ -1,11 +1,88 @@
...
+}
+
+func interfaceSlice(slice interface{}) []interface{} {
+	s := reflect.ValueOf(slice)
+	if s.Kind() != reflect.Slice {
+		panic("InterfaceSlice() given a non-slice type")
+	}
+
+	// Keep the distinction between nil and empty slice input
+	if s.IsNil() {
```
Comment (ericcheatham) [inline utils/display/display.go:44]: I'm not sure you'd ever hit this case. If the parameter `slice` is `nil` the conditional `if s.Kind() != reflect.Slice` evaluates to true.

File: utils/display/environments_test.go
Comment (janelletavares) [reply utils/display/environments_test.go:1]: https://www.notion.so/meroxa/MDPX-mini-platform-1ef6301678b04b87a394b04e18afcd1a?pvs=4#4b92ba9f2a01446ab359b779185ecb3a it's under requirements

File: utils/display/display.go (hunk 3 above)
Comment (janelletavares) [reply utils/display/display.go]: i made some changes and tested with with a hacked up unit test

**Review by ericcheatham — APPROVED** [review]

**Review by anna-cross — APPROVED** [review]

**Comment by janelletavares** [comment]
I pushed everything up to github that I had worked on

**Pull request merged**